pkgs/by-name/yt/yt-dlp/package.nix

```
@@ -1,32 +1,90 @@
 {
   lib,
   python3Packages,
+  atomicparsley,
+  deno,
   fetchFromGitHub,
+  fetchNpmDeps,
   ffmpeg-headless,
-  rtmpdump,
-  atomicparsley,
-  pandoc,
   installShellFiles,
+  nodejs,
+  npmHooks,
+  pandoc,
+  rtmpdump,
   atomicparsleySupport ? true,
   ffmpegSupport ? true,
+  javascriptSupport ? true,
   rtmpSupport ? true,
   withAlias ? false, # Provides bin/youtube-dl for backcompat
   nix-update-script,
 }:
 
+let
+  yt-dlp-ejs = python3Packages.buildPythonApplication rec {
```

Why not use a separate package for this, i.e., `python3Packages.yt-dlp-ejs`?
I know we want to update yt-dlp-ejs every time we update yt-dlp but using `let` doesn't make that any easier.

---

I'm not convinced this should block this PR, but I 100% agree that this should be exposed as a separate package.

---

At least it would make sense to use `callPackage` with a separate file.
Also,
```suggestion
  yt-dlp-ejs = python3Packages.buildPythonPackage rec {
```

---

> Shouldn't this be considered an implementation detail of yt-dlp?

I consider libraries that are only used by one dependent to still be libraries, and they are worth exposing for similar reasons to why `overrideAttrs` use is discouraged. Also it results in cleaner code.

Regardless, this is definitely a nitpick and shouldn't be blocking.

---

FWIW the upstream project more or less also [recommends](https://github.com/yt-dlp/yt-dlp/issues/15012) packaging yt-dlp-ejs as separate python package.

> If yt-dlp is packaged as a Python package in your repository, yt-dlp-ejs would ideally be packaged separately

But also (as already noted), yt-dlp and yt-dlp-ejs have to be updated at the same time

> Each version of yt-dlp will be pinned to a specific version of yt-dlp-ejs and yt-dlp will reject any other yt-dlp-ejs version. Refer to yt-dlp's [pyproject.toml](https://github.com/yt-dlp/yt-dlp/blob/2025.11.12/pyproject.toml#L59) for the pinned version

---

> FWIW the upstream project more or less also [recommends](https://github.com/yt-dlp/yt-dlp/issues/15012) packaging yt-dlp-ejs as separate python package.

It is a separate python package already in a let in because it is built in a separate derivation.
They just hint you to not combine the build of both into one package. I don't know why anyone would want to do that but oh well.